lib/conn_parser.ex

```diff
@@ -1,62 +1,80 @@
 defmodule Xcribe.ConnParser do
+  @moduledoc ~S"""
+  Used to convert a `Plug.Conn` to a `Xcribe.Request`.
+
+  Each connection sent to documenting in your tests is given to `ConnParser`.
+  Is expected that connection has been passed through the app `Endpoint` as a
+  finished request. The parser will extract all needed info from `Conn` and uses
+  app `Router` for additional information about the request.
+
+  The atribute `description` must be given at `document` macro call with the
+  option `:as`:
+
+      test "test name", %{conn: conn} do
+        ...
+
+        Xcribe.Helpers.Document.document(conn, as: "description here")
+
+        ...
+      end
+
+  If no description is given the current test description will be used.
+  """
+
   alias Xcribe.{Config, Request}
 
-  def execute(conn, description \\ "sample request") do
-    route = identify_route(conn)
-    path = format_path(route.path, Map.keys(conn.path_params))
-    namespaces = fetch_namespaces()
+  @doc """
+  Parse the given `Plug.Conn` and transform it to a `Xcribe.Request`. A
+  description can be provided.
+  """
+  def execute(conn, description \\ "") do
+    conn
+    |> identify_route()
+    |> parse_conn(conn, description)
+  end
+
+  defp parse_conn({:error, _} = error, _conn, _description), do: error
+
+  defp parse_conn(route, conn, description) do
+    path = format_path(route.route, Map.keys(conn.path_params))
 
     %Request{
       action: route |> router_options() |> Atom.to_string(),
       header_params: conn.req_headers,
-      controller: conn |> controller_module(),
+      controller: controller_module(route),
       description: description,
       params: conn.params,
       path: path,
       path_params: conn.path_params,
       query_params: conn.query_params,
       request_body: conn.body_params,
-      resource: resource_name(path, namespaces),
+      resource: resource_name(path, fetch_namespaces()),
       resource_group: resource_group(route),
       resp_body: conn.resp_body,
       resp_headers: conn.resp_headers,
       status_code: conn.status,
-      verb: Atom.to_string(route.verb)
+      verb: String.downcase(conn.method)
     }
   end
 
-  defp identify_route(conn) do
+  defp identify_route(%{method: method, host: host, path_info: path} = conn) do
     conn
-    |> router_module()
-    |> apply(:__routes__, [])
-    |> enum_find(conn)
+    |> router_module
```

Comment:
```suggestion
    |> router_module()
```
:nail_care: 
Per [Elixir Style Guide](https://github.com/christopheradams/elixir_style_guide#parentheses-pipe-operator)